packages/runtime/container-runtime/src/summaryStorageCompressionHooks.ts

```diff
@@ -0,0 +1,131 @@
+     * The header is then addded which shows, which algorithm was used for encryption.
+     * The result binary is base64 encoded due to the issue with summaryTreeUploadManager#writeSummaryBlob
+     * hash assertion
+     * New ISummaryBlob is created with the new string content obtained in the above step and the
+     * old blob is replaced by this new blob in the ISummaryTree.
+     */
+    public onPreUploadSummaryWithContext(summary: ISummaryTree, context: ISummaryContext):
+    { prepSummary: ISummaryTree; prepContext: ISummaryContext; } {
+        const newSummary = cloneSummary(summary);
+        const paths: string[][] = [];
+        const currentPath: string[] = [];
+        listBlobPaths(paths, currentPath, newSummary);
+        paths.forEach((path) => {
+            const summaryBlob: ISummaryBlob = getBlobAtPath(summary, path);
+            let decompressed: Uint8Array;
+            if (typeof summaryBlob.content === "string") {
+                decompressed = new TextEncoder().encode(summaryBlob.content);
+            } else {
+                decompressed = summaryBlob.content;
+            }
+            const compressed: ArrayBufferLike = this.encodeBlob(decompressed);
+            // TODO: This step is now needed, it looks like the function summaryTreeUploadManager#writeSummaryBlob
+            // fails on assertion at 2 different generations of the hash which do not lead to
+            // the same result if the ISummaryBlob.content is in the form of ArrayBufferLike
+            const compressedString = Uint8ArrayToString(Buffer.from(compressed), "base64");
+            const compressedEncoded = new TextEncoder().encode(compressedString);
+            const newSummaryBlob: ISummaryBlob = { type: SummaryType.Blob, content: compressedEncoded };
+            replaceSummaryObject(newSummary, path, newSummaryBlob);
+        });
+        return { prepSummary: newSummary, prepContext: context };
+    }
+    /**
+     * TODO: This method is not yet implemented
+     */
+    public onPostGetSnapshotTree(tree: ISnapshotTree | null): ISnapshotTree | null {
+        return tree;
+    }
+    /**
+     * TODO: This method is not yet implemented
+     */
+    public onPostDownloadSummary(summary: ISummaryTree): ISummaryTree {
+        return summary;
+    }
+
+    private encodeBlob(file: ArrayBufferLike): ArrayBufferLike {
+        let compressed: ArrayBufferLike;
+        if (this._algorithm === Algorithms.NONE) {
+            return file;
+        } else
+            if (this._algorithm === Algorithms.DEFLATE) {
+                compressed = deflate(file) as ArrayBufferLike;
+            } else
+                if (this._algorithm === Algorithms.LZ4) {
+                    compressed = compress(file) as ArrayBufferLike;
+                } else {
+                    throw Error(`Unknown Algorithm ${this._algorithm}`);
+                }
+        const headerBuilder: BlobHeaderBuilder = new BlobHeaderBuilder();
```

Comment:
Nit: ESLint will/should require {} on all 'else' clauses.

Reply:
implemented in 51a1a8aa0003cd567a36a8145a900153bdd6a074